packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -196,6 +196,49 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
     }
   }
 
+  /**
+   * Finds the next valid Dropdown option
+   * @param stepValue Value of how many items the function should traverse.  Should be -1 or 1.
+   * @param index Index of where the search should start
+   * @param selectedIndex The selectedIndex Dropdown's state
+   * @returns The next valid dropdown option's index
+   */
+  private _moveIndex(stepValue: number, index: number, selectedIndex: number): number {
+    // Return selectedIndex if nothing has changed
+    if (selectedIndex === index) {
+      return selectedIndex;
+    }
+    const { options } = this.props;
+    let stepCounter = 0;
+
+    // Set starting index to 0 if index is < 0
+    if (index < 0) {
+      index = 0;
+    }
+    // Set starting index to last option index if greater than options.length
+    if (index >= options.length) {
+      index = options.length - 1;
+    }
+    // If current index is a header or divider, increment by step
+    while (options[index].itemType === DropdownMenuItemType.Header
+      || options[index].itemType === DropdownMenuItemType.Divider) {
+
+      // If stepCounter exceeds length of options, then return selectedIndex (-1)
+      if (stepCounter >= options.length) {
+        return selectedIndex;
+      }
+      // If index + stepValue is out of bounds, reverse step direction
+      if (index + stepValue < 0 || index + stepValue >= options.length) {
```

Comment:
This looks like danger zone here... maybe you should remove this. It is unclear why you would need to reverse the step.